File: SDKs/PopcornNativeSDK/PopcornCore/Domain/Models/Filter/SortOption.swift
//
//  SortOption.swift
//  Popcorn
//
//  Created by Eduardo Raupp Peretto on 13/05/25.
//

import Foundation

public enum SortOption: String, Equatable {
    case popularityDesc = "popularity.desc"
    case releaseDateDesc = "release_date.desc"
    case voteAverageDesc = "vote_average.desc"

    public var description: String {
        switch self {
        case .popularityDesc: "Descending popularity"
        case .releaseDateDesc: "Descending release date"
        case .voteAverageDesc: "Descending vote rating"
        }
    }
}

extension SortOption: CaseIterable {
    public static var allCases: [SortOption] = [.popularityDesc, .releaseDateDesc, .voteAverageDesc]
}

File: KotlinShared/shared/src/commonMain/kotlin/com/eduardoperetto/popcornshared/domain/SortOption.kt
package com.eduardoperetto.popcornshared.domain

// Sorting and Filtering
enum class SortOption(val value: String, val description: String) {
    popularityDesc("popularity.desc", "Descending popularity"),
    releaseDateDesc("release_date.desc", "Descending release date"),
    voteAverageDesc("vote_average.desc", "Descending vote rating");
}